### PR TITLE
Made amount optional

### PR DESCRIPTION
**Pull request by Robert-SD**

### CODE REVIEW

File: android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt
Comment (Robert-SD) [inline android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt:1]: Generated code - no review required.

File: ios/Classes/PlatformApi.swift
Comment (Robert-SD) [inline ios/Classes/PlatformApi.swift:1]: Generated code - no review required.

File: lib/src/generated/platform_api.g.dart
Comment (Robert-SD) [inline lib/src/generated/platform_api.g.dart:1]: Generated code - no review required.

**Review by atmamont — DISMISSED** [review]

File: ios/Classes/components/applePay/ApplePaySessionComponent.swift [hunk 1]
```diff
@@ -31,14 +28,13 @@ class ApplePaySessionComponent: BaseApplePayComponent {
...
-        else {
-            return nil
-        }
-        applePayComponent.delegate = sessionHolder.session
+    private func buildApplePaySessionComponent() throws -> ApplePayComponent? {
+        guard let session = sessionHolder.session else { throw PlatformError(errorDescription: "Session is not available.") }
+        guard let paymentMethod = session.sessionContext.paymentMethods.paymentMethod(ofType: ApplePayPaymentMethod.self) else { throw PlatformError(errorDescription: "Apple Pay payment method not valid.") }
+        let context = try configuration.createAdyenContext()
+        let configuration = try configuration.mapToApplePayConfiguration(payment: session.sessionContext.payment ?? context.payment)
+        let applePayComponent = try? ApplePayComponent(paymentMethod: paymentMethod, context: context, configuration: configuration)
```
Comment (goergisn) [inline ios/Classes/components/applePay/ApplePaySessionComponent.swift:36]: Is it still valid that this try does not throw?

File: ios/Classes/utils/ConfigurationMapper.swift [hunk 2]
```diff
@@ -2,17 +2,16 @@
...
         )
-        
-        dropInConfiguration.paymentMethodsList.allowDisablingStoredPaymentMethods =
-            dropInConfigurationDTO.isRemoveStoredPaymentMethodEnabled
-        
-        if let cardConfigurationDTO = dropInConfigurationDTO.cardConfigurationDTO {
+
+        dropInConfiguration.paymentMethodsList.allowDisablingStoredPaymentMethods = isRemoveStoredPaymentMethodEnabled
+
+        if let cardConfigurationDTO {
```
Comment (goergisn) [inline ios/Classes/utils/ConfigurationMapper.swift:14]: what about moving the content of this if to a different function? `func cardConfiguration(from dto:) -> CardComponent.Configuration`

**Review by goergisn — DISMISSED** [review]

File: ios/Classes/components/applePay/ApplePaySessionComponent.swift (hunk 1 above)
Comment (Robert-SD) [reply ios/Classes/components/applePay/ApplePaySessionComponent.swift:36]: Good catch!

**Review by goergisn — APPROVED** [review]

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**